src/ERC20Reserve.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.7;
+
+import {Ownable} from "openzeppelin-contracts/access/Ownable.sol";
+import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
+
+interface IERC20Reserve {
+    function withdraw(uint256 amt) external;
+
+    function deposit(uint256 amt) external;
+}
+
+contract ERC20Reserve is IERC20Reserve, Ownable {
+    IERC20 public immutable erc20;
+    address public user;
+    uint256 public balance;
+
+    event Withdrawn(address to, uint256 amt);
+    event Deposited(address from, uint256 amt);
+    event ForceWithdrawn(address to, uint256 amt);
+    event UserSet(address oldUser, address newUser);
+
+    constructor(
+        IERC20 _erc20,
+        address owner,
+        address _user
+    ) {
+        erc20 = _erc20;
+        transferOwnership(owner);
+        setUser(_user);
+    }
+
+    modifier onlyUser() {
+        require(_msgSender() == user, "Reserve: caller is not the user");
+        _;
+    }
+
+    function withdraw(uint256 amt) public override onlyUser {
```

Comment:
The `withdraw` would work with `withdraw(address usr, uint256 amt)` but it would break the consistency. :)

Reply:
Without going over the hub

Reply:
Yeah, I'm not a fan of any of these solutions either